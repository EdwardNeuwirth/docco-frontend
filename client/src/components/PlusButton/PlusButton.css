.plus-button {
  border: solid;
  border-color: #BDC3C7;
  border-width: thin;
  border-radius: 50%;
  display: flex;
  justify-content: center;
  align-items: center;
  background-color: #ECF0F1;
  transition: 500ms;
}

.plus-button:hover {
  transition: 500ms;
  border-color: #F1C40F;
  background-color: #F1C40F;
  transform: rotate(90deg)
}