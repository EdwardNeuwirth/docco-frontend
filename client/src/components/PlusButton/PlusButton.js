import React from 'react';

import './PlusButton.css';



const PlusButton = (props) => {

  const {banana} = props;
  let {size} = props;

  if(!size) size=8;


  const myStyle = {
    width: `${size}vw`,
    height: `${size}vw`,
    fontSize: `${size}vw`,
    padding: `${size*0.1}vw`
  }

  return (
    <div className="plus-button" style={myStyle} tabIndex="0" role="button" onClick={banana} onKeyPress={banana}><i className="fas fa-plus" /></div>
  )

}

export default PlusButton;