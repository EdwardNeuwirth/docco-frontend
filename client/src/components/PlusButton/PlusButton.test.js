import React from 'react';
import { shallow, mount } from 'enzyme';
import PlusButton from './PlusButton';



describe('<PlusButton />', () => {
  let wrapper;
  const click = jest.fn();
  wrapper = shallow(<PlusButton size='a' />);
    it('should render a div', () => {
      expect(wrapper.find('div').length).toEqual(1);
    });
    it('should make use of font awesome', () => {
      expect(wrapper.find('.fa-plus').length).toEqual(1);
    });
    it('should render correctly', () => {
      expect(wrapper).toMatchSnapshot();
    });
    it('should have a click prop containing a function', () => {
      expect(wrapper.prop('size')).toEqual('a');
    });

});