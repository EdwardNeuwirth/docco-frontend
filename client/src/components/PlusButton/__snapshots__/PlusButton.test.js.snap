// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PlusButton /> should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PlusButton
    size="a"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <i
        className="fas fa-plus"
      />,
      "className": "plus-button",
      "onClick": undefined,
      "onKeyPress": undefined,
      "role": "button",
      "style": Object {
        "fontSize": "avw",
        "height": "avw",
        "padding": "NaNvw",
        "width": "avw",
      },
      "tabIndex": "0",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "className": "fas fa-plus",
      },
      "ref": null,
      "rendered": null,
      "type": "i",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <i
          className="fas fa-plus"
        />,
        "className": "plus-button",
        "onClick": undefined,
        "onKeyPress": undefined,
        "role": "button",
        "style": Object {
          "fontSize": "avw",
          "height": "avw",
          "padding": "NaNvw",
          "width": "avw",
        },
        "tabIndex": "0",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "fas fa-plus",
        },
        "ref": null,
        "rendered": null,
        "type": "i",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
